rfcs/038-unify-shared-dependencies.md

```diff
@@ -0,0 +1,125 @@
+- Feature Name: Unify shared dependencies
+- Start Date: 2021-05-22
+- RFC PR: [amundsen-io/rfcs#38](https://github.com/amundsen-io/rfcs/pull/38) (after opening the RFC PR, update this with a link to it and update the file name)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Unify shared dependencies
+
+## Summary
+
+Major motivation behind migrating to monorepo was to solve `Dependency Management` issue. However, it still remains unsolved.
+
+This RFC serves as a platform to figure out the best approach for unifying shared dependencies and contains preliminary proposal for how that could be solved.
+
+## Motivation
+
+Following original RFC:
+
+```
+Dependency Management: Python packages are pretty much the same for each proxy, which makes it hard to sync across all the above repositories. As a result most of the time, each proxy is running its own version of the dependency.
+A few examples:
+
+| Package             | amundsenmetadata   | amundsenfrontend  |
+| --------------      |:------------------:| -----------------:|
+| amundsen-common     | >=0.8.1            | ==0.6.0           |
+| flake8              | ==3.5.0            | ==3.8.4           |
```

Comment:
For test libraries as flake8 I would highly recommend using something like [pre-commit](https://pre-commit.com). This makes it very easy to do the checks, and isolate the test libraries like flake8, mypy, black, etc from the rest. This would also remove the issue of test-dependencies clashing because pre-commit will create a separate environment per hook. Updating would be quite straightforward, using `pre-commit autoupdate`.

Reply:
I agree that pre-commit could be a good fit. Would you then replace `make test` in ci/cd with `pre-commit run --all-files` ?

Reply:
I think triggering `make test` would be a nice transition. Check if `pre-commit` is on the PATH, and if not, give the developer instructions on how to install this. I agree that this might be bigger than the RFC. I don't want to blow up this RFC :)

Reply:
I'll extend the list of future possibilities with this suggestion then!